packages/x-components/src/x-modules/tagging/components/__tests__/tagging.spec.ts

```diff
@@ -0,0 +1,116 @@
+import { mount, Wrapper } from '@vue/test-utils';
+import Vue from 'vue';
+import { Component, Prop, Provide } from 'vue-property-decorator';
+import { installNewXPlugin } from '../../../../__tests__/utils';
+import { getXComponentXModuleName, isXComponent } from '../../../../components/x-component.utils';
+import { XComponentBusAPI } from '../../../../plugins/x-plugin.types';
+import { baseSnippetConfig } from '../../../../views/base-config';
+import { SnippetConfig } from '../../../../x-installer/api/api.types';
+import Tagging from '../../../tagging/components/tagging.vue';
+
+@Component({
+  template: `<div><slot/></div>`
+})
+class Provider extends Vue {
+  @Prop({ default: baseSnippetConfig })
+  @Provide()
+  public snippetConfig!: SnippetConfig;
+}
+
+function renderTagging({
```

Comment:
You don't need the `Provider` component, you can do that provide in the wrapper that you are creating. Also, I would suggest to also make the `snippetConfig` reactive here, and clone it just to avoid mutating the `baseSnippetConfig` object.

```ts
function renderTagging({
  template = `<Tagging :consent="consent"/>`,
  consent = undefined,
  snippetConfig = { ...baseSnippetConfig }
}: RenderTaggingOptions = {}): RenderTaggingAPI {
  const [, localVue] = installNewXPlugin();
  snippetConfig = localVue.observable(snippetConfig);

  const wrapper = mount(
    {
      components: {
        Tagging
      },
      props: ['consent', 'snippetConfig'],
      template
    },
    {
      provide: {
        snippetConfig
      },
      propsData: {
        consent,
        snippetConfig
      },
      localVue
    }
  );

  return {
    wrapper: wrapper.findComponent(Tagging),
    on: wrapper.vm.$x.on.bind(wrapper.vm.$x)
  };
}
```